tilelang/carver/roller/shape_inference/tir.py

```
@@ -242,8 +243,8 @@ def construct_dependency_target(self, targets: Tuple[str]):
         return input_vars, mapping
 
     def infer(self,
-              shape: Dict[str, List[arith.ConstIntBound]],
-              rstep: Dict[str, int] = None,
+              shape: dict[str, list[arith.ConstIntBound]],
+              rstep: dict[str, int] = None,
```

Ruff (0.14.1)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)